modulus/models/graphcast/graph_cast_net.py

```diff
@@ -42,6 +43,119 @@
 
 from .graph_cast_processor import GraphCastProcessor
 
+logger = logging.getLogger(__name__)
+
+
+def get_lat_lon_partition_separators(partition_size: int):
+    """Utility Function to get separation intervals for lat-lon
+    grid for partition_sizes of interest.
+
+    Parameters
+    ----------
+    partition_size : int
+        size of graph partition
+    """
+
+    def _divide(num_lat_chunks: int, num_lon_chunks: int):
+        # divide lat-lon grid into equally-sizes chunks along both latitude and longitude
+        if (num_lon_chunks * num_lat_chunks) != partition_size:
+            raise ValueError(
+                "Can't divide lat-lon grid into grid {num_lat_chunks} x {num_lon_chunks} chunks for partition_size={partition_size}."
+            )
+        # divide latitutude into num_lat_chunks of size 180 / num_lat_chunks
+        # divide longitude into chunks of size 360 / (partition_size / num_lat_chunks)
+        lat_bin_width = 180.0 / num_lat_chunks
+        lon_bin_width = 360.0 / num_lon_chunks
```

Comment:
This is a good solution for the inputs which are on a lat-lon grid, but is this ideal for the icosahedral grid as well once you have `num_lat_chunks > 2`? Since the poles will have much fewer points in the icosahedral grid right? One easy solution for this would be to add a `use_cosine_lat_scaling=False` option in this method and partition based on `cos(lat)` in `[-1,1]` which would be a more even split for the icosahedral grid.

Reply:
In this way, you are correct. We might have to explore these things much more anyways. The thing is, as of now, `num_lat_chunks` only occurs for `partition_size > 8` anyways in which case the number of points per partition for GraphCast probably will be too small anyways compared to necessary communication within the processor across multiple GPU nodes. In this case, we actually might even look at letting some GPUs sit idle and use a smaller partition for the latent mesh itself.

The consideration here is also a different one. The "edges" between the grid and the mesh, mainly are based on e.g. nearest-neighbor relations of grid points on the lat-lon grid and the processor mesh.
By using the same partitioning throughout, you can avoid the processor nodes having to "gather" grid nodes from other partitions, as most of the mesh-nodes' neighbors are within the same lat-lon chunks.

Reply:
Yeah, makes sense that this is only needed for `partition_size > 8`. Let's add a comment here so we know what to do if this is needed down the line?